src/pasta/fields/fp.rs

```diff
@@ -781,11 +773,36 @@ fn test_inv() {
     assert_eq!(inv, INV);
 }
 
+#[test]
+fn test_rescue() {
+    // NB: TWO_INV is standing in as a "random" field element
+    assert_eq!(
+        Fp::TWO_INV
+            .pow_vartime(&[Fp::RESCUE_ALPHA, 0, 0, 0])
+            .pow_vartime(&Fp::RESCUE_INVALPHA),
+        Fp::TWO_INV
+    );
+}
+
+#[test]
+fn test_sqrt() {
+    // NB: TWO_INV is standing in as a "random" field element
+    let v = (Fp::TWO_INV).square().sqrt().unwrap();
+    assert!(v == Fp::TWO_INV || (-v) == Fp::TWO_INV);
+}
+
+#[test]
+fn test_deterministic_sqrt() {
+    // NB: TWO_INV is standing in as a "random" field element
+    let v = (Fp::TWO_INV).square().deterministic_sqrt().unwrap();
+    assert!(v == Fp::TWO_INV || (-v) == Fp::TWO_INV);
```

Comment:
This should test the determinism and assert which sign the result has.

Reply:
See https://github.com/zcash/halo2/pull/84#discussion_r541155304